internal/go_repository.bzl

```diff
@@ -33,6 +52,7 @@ def _go_repository_impl(ctx):
             sha256 = ctx.attr.sha256,
             stripPrefix = ctx.attr.strip_prefix,
             type = ctx.attr.type,
+            auth = _get_auth(ctx, ctx.attr.urls),
```

Comment:
Have you verified this works? Looks like we are setting the `auth` param but not using it when we download.

Reply:
It works. I believe `ctx.download_and_extract` uses the auth param